daphne/src/taskprov.rs

```diff
@@ -0,0 +1,35 @@
+use crate::{
+    messages::{
+        taskprov::{VdafType, TASK_PROV_SALT},
+        Id,
+    },
+    vdaf::VdafVerifyKey,
+};
+use ring::{
+    error::Unspecified,
+    hkdf::{Prk, Salt, HKDF_SHA256},
+};
+
+pub fn compute_vdaf_verify_prk(verify_key_init: &[u8]) -> Prk {
+    // The documentation says computing the Salt is expensive, and we use the same PRK all the
+    // time, so we compute it once.
+    Salt::new(HKDF_SHA256, &TASK_PROV_SALT).extract(verify_key_init)
+}
+
+/// Compute the VDAF verify key for task_id and the specified VDAF type.
+pub fn compute_vdaf_verify_key(
+    prk: &Prk,
+    task_id: &Id,
+    vdaf_type: VdafType,
+) -> Result<VdafVerifyKey, Unspecified> {
+    let info = [task_id.as_ref()];
+    let okm = prk.expand(&info, vdaf_type.clone())?;
```

Comment:
Perhaps I'm reading this wrong, but this doesn't look quite right. [The spec says](https://datatracker.ietf.org/doc/html/draft-wang-ppm-dap-taskprov-00#section-3.2):

```
   verify_key = HKDF-Expand(
       HKDF-Extract(
           task_prov_salt,  # salt
           verify_key_init, # IKM
       ),
       task_id,             # info
       VERIFY_KEY_SIZE,     # L
   )
```

So I would expect to see a call to [prk = extract(task_prov_salt, verify_key_init)](https://docs.rs/ring/0.14.6/ring/hkdf/fn.extract.html) then to expand(prk, task_id, vdafy_type.len()).

Reply:
The HKDF docs say that making a salt is kind of expensive, and since we use the same extracted value over and over, I split the "extract" part from the expand part.  The extract() is done when making the PRK in compute_vdaf_verify_prk(), and then we use that prk as a parameter in all the calls to compute_vdaf_verify_key().

Reply:
Also, the API is funky in that expand() doesn't actually expand, it merely configures expansion and returns an Okm object, on which you then call fill() to actually generate bytes.

Reply:
OH yes, thanks I missed that bit. I was also thrown by passing vdafy_type to expand. This looks right.